geonode/static/geonode/js/upload/LayerInfo.js

```diff
@@ -706,10 +706,14 @@ define(function (require, exports) {
                     layer_name = target.data('layer'),
                     file_name  = target.data('file');
                 self.removeFile(file_name);
+                if (self.files.length == 0) {
+                    delete layers[self.name];
+                }
                 if (file_ext === 'xml') {
                     $('#metadata_uploaded_preserve_check').hide();
                 }
-                self.displayRefresh();
+                self.errors = self.collectErrors();
+                self.displayErrors();
```

Comment:
Displaying files is now being handled in upload.js (L355), so I've extracted only the error collection steps.